src/hotspot/share/runtime/statSampler.cpp

```diff
@@ -260,6 +260,10 @@ void StatSampler::create_system_property_instrumentation(TRAPS) {
   add_property_constant(JAVA_PROPERTY, "java.library.path", CHECK);
   add_property_constant(JAVA_PROPERTY, "java.class.path", CHECK);
   add_property_constant(JAVA_PROPERTY, "java.home", CHECK);
+
+  add_property_constant(JAVA_PROPERTY, "jdk.module.path", CHECK);
+  add_property_constant(JAVA_PROPERTY, "jdk.module.upgrade.path", CHECK);
+  add_property_constant(JAVA_PROPERTY, "jdk.module.main", CHECK);
```

Comment:
These are not "java" properties - do we need to add a "jdk" namespace?

Reply:
correct @dholmes-ora however a couple of points to note:

- these properties *are* listed in the javadoc for System::getProperties also "jdk.debug" is a pre-existing property also present in the JAVA_PROPERTY NS ... which somewhat sets an awkward precedent...

happy to add a new JDK_PROPERTY NS but given the pre-existence of "jdk.debug" not clear to me that such would not introduce an arbitrary inconsistency

thoughts?

Reply:
I'm not sure what the list of system properties in `System.get_properties` has to do with this. That list includes a bunch of properties in the `java` namespace and a bunch not.

The pre-existence of jdk.debug is unfortunate and, I think, a mistake. As you can see from the JBS issue description the original intent was to have java.vm.debug but it got changed to jdk.debug:

https://bugs.openjdk.org/browse/JDK-8139986

I think the question that needs to be answered is why do these different namespaces even exist? AFAICT the SUN_* namespace is used for internal/implementation things and arguably if we could have we might have renamed that to JDK_*. But the reality is that we are dealing with an archaic part of the system that really reflects an age long past and could do with some modernization.

So I guess for the sake of moving forward we can ignore the namespace for now and just flag these as "Java" properties.